x-pack/plugins/uptime/public/lib/helper/component_test_helpers.tsx

```diff
@@ -0,0 +1,187 @@
+import { Router } from 'react-router-dom';
+import { MemoryHistory } from 'history/createMemoryHistory';
+import { createMemoryHistory, History } from 'history';
+import { I18nProvider } from '@kbn/i18n/react';
+import { mountWithIntl, renderWithIntl, shallowWithIntl } from '@kbn/test/jest';
+import { coreMock } from 'src/core/public/mocks';
+import {
+  KibanaContextProvider,
+  KibanaServices,
+} from '../../../../../../src/plugins/kibana_react/public';
+import { MountWithReduxProvider } from './helper_with_redux';
+import { AppState } from '../../state';
+
+interface KibanaProps {
+  services?: KibanaServices;
+}
+
+interface KibanaProviderOptions {
+  coreOptions?: any;
+  kibanaProps?: KibanaProps;
+}
+
+interface MockKibanaProviderProps extends KibanaProviderOptions {
+  children: ReactElement;
+}
+
+interface MockRouterProps extends MockKibanaProviderProps {
+  history?: History;
+}
+
+interface RenderRouterOptions extends KibanaProviderOptions {
+  history?: History;
+  renderOptions?: any;
+  state?: any;
+}
+
+/* default mock core */
+const defaultCore = coreMock.createStart();
+const mockCore: () => any = () => {
+  const core = {
+    ...defaultCore,
+    application: {
+      getUrlForApp: () => '/app/uptime',
+      navigateToUrl: jest.fn(),
+    },
+  };
+
+  return core;
+};
+
+/* Higher Order Components */
+export function withKibanaContext<T>(
+  WrappedComponent: React.ComponentType<T>,
+  { kibanaProps, coreOptions }: KibanaProviderOptions
+) {
+  const core = {
+    ...mockCore(),
+    coreOptions,
+  };
+  return (props: any) => (
+    <KibanaContextProvider services={{ ...core }} {...kibanaProps}>
+      <I18nProvider>
+        <WrappedComponent {...(props as T)} />
+      </I18nProvider>
+    </KibanaContextProvider>
+  );
+}
+
+export function withRouter<T>(WrappedComponent: React.ComponentType<T>, customHistory: History) {
+  const history = customHistory || createMemoryHistory();
+  return (props: T) => (
+    <Router history={history}>
+      <WrappedComponent {...(props as T)} />
+    </Router>
+  );
+}
+
+/* Mock Provider Components */
+export function MockKibanaProvider({
+  children,
+  coreOptions,
+  kibanaProps,
+}: MockKibanaProviderProps) {
+  const core = {
+    ...mockCore(),
+    coreOptions,
+  };
+  return (
+    <KibanaContextProvider services={{ ...core }} {...kibanaProps}>
+      <I18nProvider>{children}</I18nProvider>
+    </KibanaContextProvider>
+  );
+}
+
+export function MockRouter({
+  children,
+  coreOptions,
+  history: customHistory,
+  kibanaProps,
+}: MockRouterProps) {
+  const history = customHistory || createMemoryHistory();
+  return (
+    <Router history={history}>
+      <MockKibanaProvider coreOptions={coreOptions} kibanaProps={kibanaProps}>
+        {children}
+      </MockKibanaProvider>
+    </Router>
+  );
+}
+
+export const render = (
+  ui: ReactElement,
+  { history, coreOptions, kibanaProps, renderOptions, state }: RenderRouterOptions = {}
+) => {
+  return reactRender(
```

Comment:
Maybe we should alias this as `reactTestRender` to denote it's calling `render` imported from RTL and not `react-dom`.

I think if the reader understands the context of this file it's probably not necessary, but some more descriptive name like that (or a better suggestion) might improve readability.

Reply:
Maybe we should name it something other than `reactTestRender` since react test renderer is a thing that it may be confused with. How about `reactTestLibRender`

Reply:
That sounds fine to me. This way it will disambiguate the source of the function for the person reading the code.